### DISCUSSION TITLE
Control time-lapse by GPS position

### DISCUSSION THREAD

**Discussion by richlv** [gopro/labs#688] [Feature Requests]
Is there a way to control time-lapse by the GPS position?
That is, have it shoot a picture every 0.5, 2 etc seconds, but pause if the position has not changed by some configurable distance in metres.

**Comment by dnewman-gpsw** [comment]
Yes, I've done this, although you are using the script to take photos, not a inbuilt timelapse mode.

This is the script I used on MAX, although it should work on HERO11 fine too.

**mP!D>c:A10oMFRMA=0,0+!S!R**

This will take a photo every 10 meters.

**Comment by richlv** [comment]
Thank you so much, is 10 in here "c:A**10**o" the distance?
When you say "not a inbuilt timelapse mode", is there any way I could start-stop this mode easily with camera buttons?

My usecase - capturing imagery for open projects, for OpenStreetMap usage - like Mapillary.
For example, I might have camera attached to a car or a bicycle, and I'd like to have it pause when not moving - but at the same time easily start/stop capture manually as well (for example, when removing the camera, entering a building etc) so that it doesn't take pictures facing ground or so. It would be perfect if I could extend one of the existing modes to operate like this, and then I could set that mode to be the default one.

Sorry if these are silly questions, never had a labs-compatible device before.

**Reply by dnewman-gpsw** [reply]
See more command details here https://gopro.github.io/labs/control/actions/

This command is pretty dense, doing a lot with limited input.

This is the core
>c:A10oMFRMA=0,0+!S!R
...
If(distance A > 10) //>c:A10
{
  Set_start_coord_to_current_location(); // oMFRMA=0,0
  StartCapture();  //!S
}
Repeat();  //!R
...

This will not prevent manual shutter captures.

**Reply by richlv** [reply]
Thank you for the explanation, very helpful.
Sorry if my description was confusing - I was thinking about a way to map any camera mode to a script like this, so I could just set that mode, press the start button and have this magic happen. Is that possible in some way?

**Reply by dnewman-gpsw** [reply]
Nothing like that, just save QR codes for the action needed
